python/pyarrow/_compute.pyx

```diff
@@ -0,0 +1,60 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# cython: language_level = 3
+
+from pyarrow.lib cimport (
+    Array,
+    wrap_datum,
+    _context,
+    check_status,
+    ChunkedArray
+)
+from pyarrow.includes.libarrow cimport CDatum, Sum
+
+
+cdef _sum_array(array: Array):
+    cdef CDatum out
+
+    with nogil:
+        check_status(Sum(_context(), CDatum(array.sp_array), &out))
+
+    return wrap_datum(out)
+
+
+cdef _sum_chunked_array(array: ChunkedArray):
+    cdef CDatum out
+
+    with nogil:
+        check_status(Sum(_context(), CDatum(array.sp_chunked_array), &out))
+
+    return wrap_datum(out)
+
+
+def sum(array):
+    """
+    Sum the values in a numerical (chunked) array.
+    """
```

Comment:
Can you expand the docstring a bit? (eg with a Parameters/Returns section, it might be useful to know it returns a  pyarrow Scalar)